src/Tooltip.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component, createRef } from 'react'
+import { createPortal } from 'react-dom'
+import propTypes from '@dhis2/prop-types'
+import { resolve } from 'styled-jsx/css'
+
+import { Popper } from './Popper.js'
+import { colors, layers } from './theme.js'
+import { referencePlacementPropType } from './common-prop-types.js'
+;('') // TODO: https://github.com/jsdoc/jsdoc/issues/1718
+
+const TOOLTIP_OFFSET = 4
+
+const popperStyle = resolve`
+    z-index: ${layers.applicationTop};
+    pointer-events: none;
+`
+
+const offsetModifier = {
+    name: 'offset',
+    options: {
+        offset: [0, TOOLTIP_OFFSET],
+    },
+}
+
+const OPEN_DELAY = 200
+const CLOSE_DELAY = 400
+
+/**
+ * @module
+ * @param {Tooltip.PropTypes} props
+ * @returns {React.Component}
+ *
+ * @example import { Tooltip } from '@dhis2/ui-core'
+ *
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/tooltip.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/tooltips--default-fluid|Storybook}
+ */
+class Tooltip extends Component {
+    state = { open: false }
+    ref = createRef()
+    openTimeout = null
+    closeTimeout = null
+
+    componentWillUnmount() {
+        clearTimeout(this.openTimeout)
+        clearTimeout(this.closeTimeout)
+    }
+
+    onMouseOver = () => {
+        clearTimeout(this.closeTimeout)
+
+        this.openTimeout = setTimeout(() => {
+            this.setState({ open: true })
+        }, OPEN_DELAY)
+    }
+
+    onMouseOut = () => {
+        clearTimeout(this.openTimeout)
+
+        this.closeTimeout = setTimeout(() => {
+            this.setState({ open: false })
+        }, CLOSE_DELAY)
+    }
+
+    render() {
+        const {
+            children,
+            className,
+            content,
+            dataTest,
+            display,
+            maxWidth,
+            placement,
+        } = this.props
+        const { open } = this.state
+
+        return (
+            <>
```

Comment:
No need for using a fragment. Both popper and the styles can be put into the `span`.

Reply:
I'm not too sure about this one, for the following reasons:
- Even though I realise that technically we are nesting a `div` in a `span`, which would result in invalid HTML, the JSX still looks a lot like that. So I think that using a fragment provides better semantics
- We are not nesting a portal within another element/component in any of our other components either. I think using this markup makes it very clear what is happening, the thing in the Portal is a separate element, not a child.

I think both arguments above are a valid reason to keep the fragment. The only valid reason to remove it imo would be if there was a substantial overhead associated with `React.Fragment` but I am not aware of any.

Looking forward to finding out what you think.

Reply:
I'm convinced! Keep it as it is